aten/src/ATen/native/cuda/TensorShape.cu

```diff
@@ -0,0 +1,385 @@
+#define TORCH_ASSERT_ONLY_METHOD_OPERATORS
+#include <ATen/Dispatch.h>
+#include <ATen/core/Tensor.h>
+#include <ATen/cuda/CUDAContext.h>
+#include <ATen/ops/empty.h>
+#include <c10/cuda/CUDAGuard.h>
+#include <c10/util/Logging.h>
+
+namespace at::native {
+
+static constexpr int64_t BLOCK_SIZE = 128;
+static constexpr int64_t BYTES_PER_THREAD = 16;
+
+namespace detail {
```

Comment:
Could you have a small block comment here that explains at a high level the strategy used here for the fused kernel. So that when we come back to it in a couple years we can easily remember :)